rcl lifecycle api changes in post-foxy rolling distro

CI builds are failing on all 3 platforms due to an api change in the rcl lifecycle api.

See:
https://github.com/ros2/rclcpp/issues/1506
https://github.com/ros2/rcl/pull/882

```
> rclnodejs@0.18.1 install /Users/distiller/RobotWebTools/rclnodejs
> node-gyp rebuild
...
  CXX(target) Release/obj.target/rclnodejs/src/rcl_bindings.o
../src/rcl_bindings.cpp:563:27: warning: overlapping comparisons always evaluate to false [-Wtautological-overlap-compare]
  if (type < RCL_ROS_TIME && type > RCL_STEADY_TIME) {
      ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
  
CXX(target) Release/obj.target/rclnodejs/src/rcl_handle.o
  CXX(target) Release/obj.target/rclnodejs/src/rcl_lifecycle_bindings.o
../src/rcl_lifecycle_bindings.cpp:79:28: error: no matching function for call to 'rcl_lifecycle_state_machine_init'
                           rcl_lifecycle_state_machine_init(
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/macros.hpp:30:49: note: expanded from macro 'THROW_ERROR_IF_NOT_EQUAL'
  CHECK_OP_AND_THROW_ERROR_IF_NOT_TRUE(!=, lhs, rhs, message)
                                                ^~~
../src/macros.hpp:22:16: note: expanded from macro 'CHECK_OP_AND_THROW_ERROR_IF_NOT_TRUE'
    if (lhs op rhs) {                                               \
               ^~~
/Users/distiller/ros2_install/ros2-osx/include/rcl_lifecycle/rcl_lifecycle.h:244:1: note: candidate function not viable: requires 9 arguments, but 10 were provided
rcl_lifecycle_state_machine_init(
^
../src/rcl_lifecycle_bindings.cpp:88:25: error: no matching function for call to 'rcl_lifecycle_state_machine_fini'
        rcl_ret_t ret = rcl_lifecycle_state_machine_fini(
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/distiller/ros2_install/ros2-osx/include/rcl_lifecycle/rcl_lifecycle.h:278:1: note: candidate function not viable: requires 2 arguments, but 3 were provided
rcl_lifecycle_state_machine_fini(
^
2 errors generated.
```

--- a/src/rcl_lifecycle_bindings.cpp
+++ b/src/rcl_lifecycle_bindings.cpp
@@ -73,14 +73,14 @@ NAN_METHOD(CreateLifecycleStateMachine) {
   const rosidl_service_type_support_t* gtg =
       GetServiceTypeSupport("lifecycle_msgs", "GetAvailableTransitions");
 
-  rcl_lifecycle_state_machine_options_t* options =
+  rcl_lifecycle_state_machine_options_t options =
       rcl_lifecycle_get_default_state_machine_options();
 
   THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
                            rcl_lifecycle_state_machine_init(
                                state_machine, node,
                                pn, cs, gs, gas, gat, gtg,
-                               options),
+                               &options),
                            rcl_get_error_string().str);
 
   auto js_obj = RclHandle::NewInstance(

--- a/src/rcl_lifecycle_bindings.cpp
+++ b/src/rcl_lifecycle_bindings.cpp
@@ -54,8 +54,6 @@ NAN_METHOD(CreateLifecycleStateMachine) {
   RclHandle* node_handle = RclHandle::Unwrap<RclHandle>(
       Nan::To<v8::Object>(info[0]).ToLocalChecked());
   rcl_node_t* node = reinterpret_cast<rcl_node_t*>(node_handle->ptr());
-  const rcl_node_options_t* node_options =
-      reinterpret_cast<const rcl_node_options_t*>(rcl_node_get_options(node));
 
   rcl_lifecycle_state_machine_t* state_machine =
       reinterpret_cast<rcl_lifecycle_state_machine_t*>(
@@ -75,18 +73,21 @@ NAN_METHOD(CreateLifecycleStateMachine) {
   const rosidl_service_type_support_t* gtg =
       GetServiceTypeSupport("lifecycle_msgs", "GetAvailableTransitions");
 
+  rcl_lifecycle_state_machine_options_t* options =
+      rcl_lifecycle_get_default_state_machine_options();
+
   THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
                            rcl_lifecycle_state_machine_init(
-                               state_machine, node, pn, cs, gs, gas, gat, gtg,
-                               true, &node_options->allocator),
+                               state_machine, node,
+                               pn, cs, gs, gas, gat, gtg,
+                               options),
                            rcl_get_error_string().str);
 
   auto js_obj = RclHandle::NewInstance(
-      state_machine, node_handle, [node, node_options](void* ptr) {
+      state_machine, node_handle, [node](void* ptr) {
         rcl_lifecycle_state_machine_t* state_machine =
             reinterpret_cast<rcl_lifecycle_state_machine_t*>(ptr);
-        rcl_ret_t ret = rcl_lifecycle_state_machine_fini(
-            state_machine, node, &node_options->allocator);
+        rcl_ret_t ret = rcl_lifecycle_state_machine_fini(state_machine, node);
         free(ptr);
         THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK, ret, rcl_get_error_string().str);
       });

